cluster-operator/src/main/java/io/strimzi/operator/cluster/model/nodepools/NodeIdRange.java

```
@@ -0,0 +1,200 @@
+     *
+     * @param range The full range as defined in the Node Pool annotation
+     *
+     * @return  Queue with the individual range sections
+     */
+    /* test */ static Queue<IdRange> parseRanges(String range)    {
+        // Remove the starting and closing brackets together with whitespaces
+        range = range.replaceAll("^\\s*\\[\\s*", "");
+        range = range.replaceAll("\\s*]\\s*$", "");
+
+        return Arrays.stream(range.split("\\s*,\\s*")).map(IdRange::new).collect(Collectors.toCollection(ArrayDeque::new));
+    }
+
+    /**
+     * Internal class to hold a single node ID or a single range
+     */
+    /* test */ static class IdRange   {
+        private Integer start;
+        private Integer end;
+
+        /**
+         * Constructs new IdRange
+         *
+         * @param rangeItem     A single node ID or range
+         *
+         * @throws InvalidNodeIdRangeException  Throws InvalidNodeIdRangeException when the range is not valid
+         */
+        public IdRange(String rangeItem) throws InvalidNodeIdRangeException {
+            if (SINGLE_NODE_ID_PATTERN.matcher(rangeItem).matches())    {
+                try {
+                    start = Integer.parseInt(rangeItem);
+                    end = null;
+                } catch (NumberFormatException e)   {
+                    throw new InvalidNodeIdRangeException("Invalid Node ID (not an integer): " + rangeItem, e);
+                }
+            } else if (RANGE_OF_NODE_IDS_PATTERN.matcher(rangeItem).matches())  {
+                String[] startAndEnd = rangeItem.split("\\s*-\\s*");
+
+                if (startAndEnd.length != 2)    {
+                    throw new InvalidNodeIdRangeException("Invalid Node ID range (exactly 2 items must be found): " + rangeItem);
+                }
+
+                try {
+                    start = Integer.parseInt(startAndEnd[0]);
+                    end = Integer.parseInt(startAndEnd[1]);
```

should we raise an exception if the user specify not consistent start and end? for example, [100-20] instead of [20-100] so using a `start` greater than `end`?

---

Ok got it. But let's make this clearer in the doc.